File: frontend/src/types/content.ts
import type { CampaignObjective, ContentIdeaStatus, ContentStatus, ContentTone } from './common'
import type { Platform } from './platform'

export interface Content {
  id: number
  title: string
  copyText: string | null
  platform: Platform
  objective: CampaignObjective | null
  status: ContentStatus
  scheduledDate: string | null
  publishedDate: string | null
  hashtags: string | null
  ctaText: string | null
  campaignId: number | null
  campaignName: string | null
  createdByName: string | null
  createdAt: string
  updatedAt: string
}

export interface ContentRequest {
  title: string
  copyText?: string
  platformId: number
  objective?: CampaignObjective
  status: ContentStatus
  scheduledDate?: string | null
  hashtags?: string
  ctaText?: string
  campaignId?: number | null
}

export interface ContentGenerationRequest {
  productOrService: string
  targetAudience: string
  platformId: number
  objective: CampaignObjective
  tone: ContentTone
  callToAction?: string
  variantCount?: number
}

export interface ContentIdea {
  id: number
  productOrService: string
  targetAudience: string | null
  platform: Platform | null
  objective: CampaignObjective | null
  tone: ContentTone
  generatedTitle: string | null
  generatedCopy: string | null
  generatedCta: string | null
  generatedHashtags: string | null
  status: ContentIdeaStatus
  createdAt: string
}

export interface ContentIdeaUpdateRequest {
  generatedTitle: string
  generatedCopy: string
  generatedCta?: string
  generatedHashtags?: string
  status?: ContentIdeaStatus
}

File: backend/src/main/java/com/marketinganalytics/platform/entity/Content.java
package com.marketinganalytics.platform.entity;

import com.marketinganalytics.platform.entity.enums.CampaignObjective;
import com.marketinganalytics.platform.entity.enums.ContentStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Instant;
import java.time.LocalDate;

/**
 * A concrete content-calendar entry (a scheduled or published post), as
 * opposed to {@link ContentIdea} which holds raw AI-generated drafts still
 * awaiting review.
 */
@Entity
@Table(name = "contents")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Content {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(name = "copy_text", columnDefinition = "TEXT")
    private String copyText;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "platform_id", nullable = false)
    private Platform platform;

    @Enumerated(EnumType.STRING)
    @Column(length = 20)
    private CampaignObjective objective;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    @Builder.Default
    private ContentStatus status = ContentStatus.IDEA;

    @Column(name = "scheduled_date")
    private LocalDate scheduledDate;

    @Column(name = "published_date")
    private LocalDate publishedDate;

    @Column(columnDefinition = "TEXT")
    private String hashtags;

    @Column(name = "cta_text", length = 150)
    private String ctaText;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "campaign_id")
    private Campaign campaign;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by")
    private User createdBy;

    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    @PrePersist
    void onCreate() {
        Instant now = Instant.now();
        createdAt = now;
        updatedAt = now;
    }

    @PreUpdate
    void onUpdate() {
        updatedAt = Instant.now();
    }
}
